instrumentation/nginx/src/otel_ngx_module.cpp

```diff
@@ -178,7 +178,16 @@ TraceContext* GetTraceContext(ngx_http_request_t* req) {
     return nullptr;
   }
 
-  return (TraceContext*)val->data;
+  std::unordered_map<ngx_http_request_t*, TraceContext*>* map = (std::unordered_map<ngx_http_request_t*, TraceContext*>*)val->data;
+  unsigned count = map->count(req);
+  if (count == 1) {
+    return map->at(req);
+  } else if (count > 1) {
+    ngx_log_error(NGX_LOG_ERR, req->connection->log, 0, "More than one TraceContext found for a request");
+    return nullptr;
+  }
+  ngx_log_error(NGX_LOG_ERR, req->connection->log, 0, "TraceContext not found");
+  return nullptr;
```

Comment:
```suggestion
auto it = map->find(req);
if (it != map->end()) {
  return it->second;
}
ngx_log_error(NGX_LOG_ERR, req->connection->log, 0, "TraceContext not found");
return nullptr;
```

Reply:
^ Just a nitpick as afaik `count` can't be greater than 1 for `unordered_map` and using find is 1 lookup instead of 2 for `count` and `at`

Reply:
@seemk You're right. I reread the documentation and that is the case.